Feat#47 유저 정보 조회 / Member 엔티티 리팩토링

kkyu0718 opened the pull request:
## 📄 작업 설명
- 유저 정보 조회 api 
- 팀 코드 스타일 확립에 따른 Member 엔티티 리팩토링

## 🚨 관련 이슈
closes #47 

## 🌈 작업 상황
- 유저 정보 조회 api 추가 완료
- Member 엔티티 리팩토링 (dto 검증, 엔티티 칼럼 상세하게 정의, dto 테스트 코드 추가)

## 📌 기타

kkyu0718 committed:
refactor: withdrawMember 인증 객체 쓰도록 변경

AuthUser 인증 객체를 받도록 수정됨에 따른 변경

kkyu0718 committed:
refactor: withdrawMember 인증 객체 쓰도록 변경

AuthUser 인증 객체를 받도록 수정됨에 따른 변경

kkyu0718 committed:
feat: MemberService 회원정보 구현

kkyu0718 committed:
feat: MemberService 회원정보 구현

kkyu0718 committed:
feat: MemberController 회원정보 조회 구현

kkyu0718 committed:
feat: MemberController 회원정보 조회 구현

kkyu0718 committed:
feat: MemberResponse 클래스 추가

회원조회 응답 클래스 추가

kkyu0718 committed:
feat: MemberResponse 클래스 추가

회원조회 응답 클래스 추가

kkyu0718 committed:
feat: 회원정보 조회 테스트 코드 추가

kkyu0718 committed:
feat: 회원정보 조회 테스트 코드 추가

kkyu0718 committed:
refactor: Member 엔티티 칼럼 정의 추가

kkyu0718 committed:
refactor: Member 엔티티 칼럼 정의 추가

kkyu0718 committed:
feat: MemberController 메소드 인자 검증 어노테이션 추가

dto 제약조건 검증 위한 @Valid 어노테이션 추가

kkyu0718 committed:
feat: MemberController 메소드 인자 검증 어노테이션 추가

dto 제약조건 검증 위한 @Valid 어노테이션 추가

kkyu0718 committed:
refactor: dto 클래스명에서 dto 제거

dto 패키지 안에 있는 것만으로도 dto 의 의미가 충분하기 때문에 제거

kkyu0718 committed:
refactor: dto 클래스명에서 dto 제거

dto 패키지 안에 있는 것만으로도 dto 의 의미가 충분하기 때문에 제거

kkyu0718 committed:
feat: MemberCreateRequest 클래스 구현과 테스트 코드 추가

- ValidationFactory 를 통해 프로퍼티 유효성 검증
- 검증에 대한 테스트 코드 추가

kkyu0718 committed:
feat: MemberCreateRequest 클래스 구현과 테스트 코드 추가

- ValidationFactory 를 통해 프로퍼티 유효성 검증
- 검증에 대한 테스트 코드 추가

HyeonJun0530 committed:
feat#48/Organization 삭제에 따른 코드 변경 (#51)

* refactor: Organization 도메인 삭제

- Organization -> Member로 1명이 관리하는 것으로 변경함에 따른 패키지 삭제

* refactor: Organization 도메인 삭제에 따른 코드 변경

- Organization이 침투한 도메인 코드 변경

* test: Organization 도메인 삭제에 따른 코드
테스트 변경

- Organization이 침투한 테스트 코드 변경

---------

Co-authored-by: HyeonJun0530 <hyjn053@gmail.com>

kkyu0718 committed:
fix: Member 엔티티 is_deleted 에 잘못 들어간 @CreationTimeStamp 어노테이션 삭제

kkyu0718 committed:
fix: Member 엔티티 is_deleted 에 잘못 들어간 @CreationTimeStamp 어노테이션 삭제

kkyu0718 committed:
fix: MemberServiceTest 잘못된 assert 문 수정

kkyu0718 committed:
fix: MemberServiceTest 잘못된 assert 문 수정

kkyu0718 committed:
Merge branch 'feat#47' of https://github.com/woowa-techcamp-2024/Team2-Dari into feat#47

HyeonJun0530 commented on src/test/java/com/wootecam/festivals/domain/member/service/MemberServiceTest.java:
```diff
@@ -105,4 +107,39 @@ void throwExceptionForNonExistentMember() {
+            MemberResponse response = memberService.findMember(memberId);
+
+            // then
+            assertNotNull(response);
+            assertEquals(memberId, response.id());
+            assertEquals("test name", response.name());
+            assertEquals("test@example.com", response.email());
+            assertEquals("test-profile-img", response.profileImg());
```
위에서는 assertAll로 작성하셨는데 여기선 빠져 있네요 !

HyeonJun0530 commented on src/test/java/com/wootecam/festivals/domain/member/service/MemberServiceTest.java:
```diff
@@ -105,4 +107,39 @@ void throwExceptionForNonExistentMember() {
+            Long nonExistentMemberId = 999L;
+
+            // when & then
+            ApiException exception = assertThrows(ApiException.class, () -> {
+                memberService.findMember(nonExistentMemberId);
+            });
+
+            assertEquals(UserErrorCode.USER_NOT_FOUND, exception.getErrorCode());
```
assertThrowBy 사용하시면 한 줄로 처리 가능해요 !

HyeonJun0530 reviewed (changes requested):
테스트 코드를 제외하곤 문제 없는 것 같습니다 👍

kkyu0718 committed:
refactor: MemberServiceTest 코드 스타일 일관성 유지되도록 수정

- 여러 개의 assert 는 assertAll 로 하도록 수정
- exception 테스트는 assertThrowBy 로 메세지까지 한번에 확인 가능하도록 수정
